lib/Reference/ProfilePickerReferenceProvider.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Andrey Borysenko <andrey18106x@gmail.com>
+ *
+ * @author 2023 Andrey Borysenko <andrey18106x@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Reference;
+
+use OC\Collaboration\Reference\LinkReferenceProvider;
+use OCP\Collaboration\Reference\ADiscoverableReferenceProvider;
+use OCP\Collaboration\Reference\Reference;
+
+use OCP\Collaboration\Reference\IReference;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+
+use OCA\Contacts\AppInfo\Application;
+use OCP\Accounts\IAccountManager;
+use OCP\IUserManager;
+
+class ProfilePickerReferenceProvider extends ADiscoverableReferenceProvider {
+
+//	private const RICH_OBJECT_TYPE = Application::APP_ID . '_profile_picker';
+	private const RICH_OBJECT_TYPE = 'users_picker_profile';
+
+	private ?string $userId;
+	private IL10N $l10n;
+	private IURLGenerator $urlGenerator;
+	private LinkReferenceProvider $linkReferenceProvider;
+	private IUserManager $userManager;
+	private IAccountManager $accountManager;
+
+	public function __construct(
+		IL10N $l10n,
+		IURLGenerator $urlGenerator,
+		LinkReferenceProvider $linkReferenceProvider,
+		IUserManager $userManager,
+		IAccountManager $accountManager,
+		?string $userId
+	) {
+		$this->userId = $userId;
+		$this->l10n = $l10n;
+		$this->urlGenerator = $urlGenerator;
+		$this->linkReferenceProvider = $linkReferenceProvider;
+		$this->userManager = $userManager;
+		$this->accountManager = $accountManager;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getId(): string	{
+		return 'profile_picker';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getTitle(): string {
+		return $this->l10n->t('Profile picker');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(): int	{
+		return 10;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getIconUrl(): string {
+		return $this->urlGenerator->imagePath(Application::APP_ID, 'profile-dark.svg');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function matchReference(string $referenceText): bool {
+		return $this->getObjectId($referenceText) !== null;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function resolveReference(string $referenceText): ?IReference {
+		if ($this->matchReference($referenceText)) {
```

Comment:
invert the conditions to have an early exit and avoid the many layers of indentation